training_data_loader.cpp

```diff
@@ -490,6 +490,57 @@ struct FeaturedBatchStream : Stream<StorageT>
 
 extern "C" {
 
+    EXPORT SparseBatch* get_sparse_batch_from_fens(
+        const char* feature_set_c,
+        int num_fens,
+        const char* const* fens,
+        int* scores,
+        int* plies,
+        int* results
+    )
+    {
+        std::vector<TrainingDataEntry> entries;
```

Comment:
Perhaps it is better to try catch the reserve method as it can throw exception and return nullptr with fprintf(stderr as it is done at the end of the function for consistency?

Reply:
If it throws we crash. If you find a realistic case where you can actually reach this (it allocates about 300 bytes per position, so you would need batches of size of millions to run out of memory) I'd be surprised.